worlds/sc2/rules.py

```diff
@@ -472,6 +542,39 @@ def the_reckoning_requirement(self, state: CollectionState) -> bool:
 
     # LotV
 
+    def protoss_defense_rating(self, state: CollectionState, zerg_enemy: bool, air_enemy: bool = True) -> int:
+        """
+        Ability to handle defensive missions
+        :param state:
+        :param zerg_enemy:
+        :param air_enemy:
+        :return:
+        """
+        defense_score = sum((pvx_defense_ratings[item] for item in pvx_defense_ratings if state.has(item, self.player)))
+        # Defensive Upgrades
+        defensive_upgrades = {item_names.KHALAI_INGENUITY, item_names.ENHANCED_TARGETING, item_names.OPTIMIZED_ORDNANCE}
+        if state.has_any({item_names.PHOTON_CANNON, item_names.KHAYDARIN_MONOLITH}, self.player):
+            defense_score += sum(item for item in defensive_upgrades if state.has(item, self.player))
+        # Corsair upgrades
+        if state.has(item_names.CORSAIR, self.player) and state.has_any({item_names.CORSAIR_ARGUS_JEWEL, item_names.CORSAIR_STEALTH_DRIVE, item_names.CORSAIR_SUSTAINING_DISRUPTION}, self.player):
+            defense_score += 1
+        # High Templar variants vs zerg
+        if state.has_any({item_names.HIGH_TEMPLAR, item_names.SIGNIFIER, item_names.ASCENDANT}, self.player) and zerg_enemy:
+            defense_score += 2
+        # Phoenix/Mirage vs air
+        if state.has_any({item_names.PHOENIX, item_names.MIRAGE}, self.player) and air_enemy:
+            defense_score += 1
```

Comment:
Nearly useless in The Dig (and vs Zerg Corsairs are just better due to splash) +1 Corsair's base armor is also better vs Mutas

I think relevant missions just need proper AA (vs light or armored)

Reply:
Phoenix and Mirage are arguably pretty good on The Dig because the major threats are Immortals and WoL Archons (which can be lifted), and eventually Colossi which are vulnerable to air, not to mention being able to shoot down the warp prisms incredibly easily... honestly, in In Utter Darkness, Phoenixes are pretty great until the enemy starts sending unliftable units.

I don't think it's that significant though, so I may just remove the air check from the protoss defense method entirely and add a comment saying it should be paired with the relevant logic rule.